lib/vrl/stdlib/src/to_regex.rs

```diff
@@ -0,0 +1,70 @@
+use regex::Regex;
+use vrl::prelude::*;
+
+#[derive(Clone, Copy, Debug)]
+pub struct ToRegex;
+
+impl Function for ToRegex {
+    fn identifier(&self) -> &'static str {
+        "to_regex"
+    }
+
+    fn parameters(&self) -> &'static [Parameter] {
+        &[Parameter {
+            keyword: "value",
+            kind: kind::BYTES,
+            required: true,
+        }]
+    }
+
+    fn examples(&self) -> &'static [Example] {
+        &[
+            Example {
+                title: "regex",
+                source: "to_regex(s'^foobar$')",
+                result: Ok("r'^foobar$'"),
+            }
+        ]
+    }
+
+    fn compile(&self, mut arguments: ArgumentList) -> Compiled {
+        let value = arguments.required("value");
+        Ok(Box::new(ToRegexFn { value }))
+    }
+}
+
+#[derive(Debug, Clone)]
+struct ToRegexFn {
+    value: Box<dyn Expression>,
+}
+
+impl Expression for ToRegexFn {
+    fn resolve(&self, ctx: &mut Context) -> Resolved {
+        let value = self.value.resolve(ctx)?.try_bytes()?;
+        let regex = regex::Regex::new(value).map_err(|err| format!("could not create regex: {}",err))?.map(Into::into);
+    }
+
+    fn type_def(&self, state: &state::Compiler) -> TypeDef {
+        self.value
+            .type_def(state)
+            .fallible_unless(
```

Comment:
This should always be fallible as the regex may be invalid.

```suggestion
            .fallible()
```